python/fusion_engine_client/__init__.py

```diff
@@ -0,0 +1,3 @@
+__all__ = ['messages', 'parsers', 'utils']
+__version__ = '1.15.0'
```

Comment:
So this is always the most annoying part of the Python release stuff, but it looks like were going with separately specifying the release tag and the version string passed to the setup.py?

I usually see some sort of file generation in the release process like we use in Nautilus for generating/updating this value.

Reply:
like to derive it from the tagname in the GA action?

Reply:
So I agree - definitely don't want to specify it in a lot of places. I actually modify `setup.py` to read this string from here (borrowed that from pyserial's setup.py file), so this + the git tag are the only two places the version is specified.

@anathan in theory I agree, best would be to just specify it once via the git tag. That way there's no risk of forgetting to update this file before a release. Unfortunately, in order for it to be set as a `__version__` variable within the loaded module at runtime, the build step would need to produce that file dynamically. I couldn't quite figure out how to get the `build` module to do that. Also, it would mean that the version string would not be set at all if anyone was doing active development and not installing a wheel from pip.

So I figured this was the best compromise for now: set the version string here, which automatically informs `setup.py`, and then also specify it in the git tag later. But I'm totally open to other ideas.

Another idea would be to write a tagging script that sets the git tag from the version string instead of the other way around like `git tag -a v$(python setup.py --version)`. Maybe we do that? (Reference: https://stackoverflow.com/questions/6786555/automatic-version-number-both-in-setup-py-setuptools-and-source-code)

Reply:
Ok, I added a new `tag_release.sh` script. @axlan @anathan do you guys mind looking that over real quick?

When we want to tag a release, we can now run `tag_release.sh 1.16.0` or similar, and it will update the Python and C++ version strings/macros (I added a `version.h` for C++), and then create a git tag.